tutorial/source/ekf.ipynb

```diff
@@ -0,0 +1,213 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Kalman Filter\n",
+    "\n",
+    "Kalman filters are linear models for state estimation of dynamic systems [1].  They have been the <i>de facto</i> standard in many robotics and tracking/prediction applications because they are well suited for systems with uncertainty about an observable dynamic process.  They use a \"observe, predict, correct\" paradigm to extract information from an otherwise noisy signal. In Pyro, we can build Kalman filters using the [`pyro.contrib.tracking` library](http://docs.pyro.ai/en/dev/contrib.tracking.html#module-pyro.contrib.tracking.extended_kalman_filter)\n",
+    "\n",
+    "## Dynamic process\n",
+    "\n",
+    "To start, consider this simple motion model:\n",
+    "\n",
+    "$$ X_{k+1} = FX_k + \\mathbf{W}_k $$\n",
+    "$$ \\mathbf{Z}_k = HX_k + \\mathbf{V}_k $$\n",
+    "\n",
+    "where $k$ is the state, $X$ is the signal estimate, $Z_k$ is the observed value at timestep $k$, $\\mathbf{W}_k$ and  $\\mathbf{V}_k$ are independent noise processes (ie $\\mathbb{E}[w_k v_j^T] = 0$ for all $j, k$) which we'll approximate as Gaussians. Note that the state transitions are linear."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Kalman Update\n",
+    "At each time step, we perform a prediction for the mean and covariance:\n",
+    "$$ \\hat{X}_k = F\\hat{X}_{k-1}$$\n",
+    "$$\\hat{P}_k = FP_{k-1}F^T + Q$$\n",
+    "and a correction for the measurement:\n",
+    "$$ K_k = \\hat{P}_k H^T(H\\hat{P}_k H^T + R)^{-1}$$\n",
+    "$$ X_k = \\hat{X}_k + K_k(z_k - H\\hat{X}_k)$$\n",
+    "$$ P_k = (I-K_k H)\\hat{P}_k$$\n",
+    "\n",
+    "where $X$ is the position estimate, $P$ is the covariance matrix, $K$ is the Kalman Gain, and $Q$ and $R$ are covariance matrices.\n",
+    "\n",
+    "For an in-depth derivation, see \\[1\\]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Nonlinear Estimation: Extended Kalman Filter\n",
+    "\n",
+    "What if our system is non-linear, eg in GPS navigation?  Consider this non-linear system:\n",
+    "\n",
+    "$$ X_{k+1} = \\mathbf{f}(X_k) + \\mathbf{W}_k $$\n",
```

Comment:
For some reason, the `f` equation is not rendering. Does it render in your local copy?

Reply:
hm it loads locally, ill build the html to double check